test/e2e/config/docker.yaml

```diff
@@ -128,3 +128,4 @@ intervals:
   default/wait-machine-remediation: ["5m", "10s"]
   node-drain/wait-deployment-available: ["3m", "10s"]
   node-drain/wait-control-plane: ["15m", "10s"]
+  node-drain/wait-machine-deleted: ["2m", "10s"]
```

Comment:
Being new to this code this name makes me thing this is the time out to wait for the machine to be fully deleted, though by reading the code I understand this is the timeout for the new machine to come up, not sure if there's a better name may be `node-drain/wait-machine-recreated`

Reply:
this _is_ the timeout waiting for a machine to be fully deleted (if you look at where it's used in the code, it's for scaling to 0 replica count for MD - and 3 to 1 for control plane)

Reply:
Tried to explain this in PR description, let me know if it's unclear:

> Now, we take a node-drain/wait-machine-deleted interval in the e2e config and calculate the appropriate machine deployment and control plane scale down timeout by adding the provided machine deletion timeout with the nodeDrainTimeout and multiply it by the number of machines.